homeassistant/components/climate/__init__.py

```diff
@@ -239,309 +185,258 @@ def state_attributes(self):
+            data[ATTR_MAX_HUMIDITY] = self.max_humidity
 
         if supported_features & SUPPORT_FAN_MODE:
-            data[ATTR_FAN_MODE] = self.current_fan_mode
-            if self.fan_list:
-                data[ATTR_FAN_LIST] = self.fan_list
+            data[ATTR_FAN_MODE] = self.fan_mode
+            data[ATTR_FAN_LIST] = self.fan_list
 
-        if supported_features & SUPPORT_OPERATION_MODE:
-            data[ATTR_OPERATION_MODE] = self.current_operation
-            if self.operation_list:
-                data[ATTR_OPERATION_LIST] = self.operation_list
+        if supported_features & SUPPORT_CURRENT_HVAC:
+            data[ATTR_CURRENT_HVAC] = self.current_hvac
 
-        if supported_features & SUPPORT_HOLD_MODE:
-            data[ATTR_HOLD_MODE] = self.current_hold_mode
+        if supported_features & SUPPORT_PRESET_MODE:
+            data[ATTR_PRESET_MODE] = self.preset_mode
+            data[ATTR_PRESET_LIST] = self.preset_list
 
         if supported_features & SUPPORT_SWING_MODE:
-            data[ATTR_SWING_MODE] = self.current_swing_mode
-            if self.swing_list:
-                data[ATTR_SWING_LIST] = self.swing_list
-
-        if supported_features & SUPPORT_AWAY_MODE:
-            is_away = self.is_away_mode_on
-            data[ATTR_AWAY_MODE] = STATE_ON if is_away else STATE_OFF
+            data[ATTR_SWING_MODE] = self.swing_mode
+            data[ATTR_SWING_LIST] = self.swing_list
 
         if supported_features & SUPPORT_AUX_HEAT:
-            is_aux_heat = self.is_aux_heat_on
-            data[ATTR_AUX_HEAT] = STATE_ON if is_aux_heat else STATE_OFF
+            data[ATTR_AUX_HEAT] = STATE_ON if self.is_aux_heat else STATE_OFF
 
         return data
 
     @property
-    def temperature_unit(self):
+    def temperature_unit(self) -> str:
         """Return the unit of measurement used by the platform."""
-        raise NotImplementedError
+        raise NotImplementedError()
 
     @property
-    def current_humidity(self):
+    def current_humidity(self) -> Optional[int]:
         """Return the current humidity."""
         return None
 
     @property
-    def target_humidity(self):
+    def target_humidity(self) -> Optional[int]:
         """Return the humidity we try to reach."""
         return None
 
     @property
-    def current_operation(self):
-        """Return current operation ie. heat, cool, idle."""
-        return None
+    def hvac_state(self) -> str:
+        """Return hvac operation ie. heat, cool mode.
+
+        Need to be one of HVAC_MODE_*.
+        """
+        raise NotImplementedError()
+
+    @property
+    def hvac_modes(self) -> List[str]:
+        """Return the list of available hvac operation modes.
+
+        Need to be a subset of HVAC_MODES.
+        """
+        raise NotImplementedError()
 
     @property
-    def operation_list(self):
-        """Return the list of available operation modes."""
+    def current_hvac(self) -> Optional[str]:
```

Comment:
It's a bit weird to add `current` since all properties represent the current state. Maybe we should call this `hvac_operation` or `hvac_action` ? (like action more I think)